management_app/views/faculty.py

```diff
@@ -1,42 +1,221 @@
+
+def process_user_template(file_path, sheet__index, sheet__index_name, action_type):
+    df = pd.read_excel(file_path, sheet_name=sheet__index or sheet__index_name)
+    df.loc[df['admin'].isnull(),'admin'] = 0 #Check if NaN in the sheet
+    df['admin'] = df['admin'].astype(int)
+    users_data = df.values.tolist()
+
+    for user in users_data:
+        user_name = user[0]
+        user_email = user[1]
+        user_ucinetid = user[2]
+        user_role = user[3]
+        is_admin = user[4]
+
+        db = get_db()
+        db.execute("PRAGMA foreign_keys = ON")
+        user = get_exist_user(db, user_ucinetid) # Check exist user by ucinetid
+
+        if user is None and action_type == 'addUsers':
+            try:
+                insert_users(db, user_name, user_email, user_ucinetid, is_admin)
+                insert_users_status(db, user_ucinetid, 2020, user_role, 1) # TODO: confirm if we can have active_status in template (temporarily set current year 2020)
+            except db.IntegrityError:
+                print('INTEGRITY ERROR\n', traceback.print_exc())
+        elif action_type == 'editUsers':
+            user_id = user['user_id']
+            try:
+                update_users(db, user_name, is_admin, user_ucinetid)
+                update_user_status_by_users_template(db, user_id, 2022, user_role) # TODO: confirm if we can have active_status in template (temporarily set current year 2020)
+            except db.IntegrityError:
+                print('INTEGRITY ERROR\n', traceback.print_exc())
+    return
+
+def process_professors_point_template(file_path, sheet__index, sheet__index_name, action_type):
+    df = pd.read_excel(file_path, sheet_name=sheet__index or sheet__index_name)
+    data = df.values.tolist()
+
+    for d in data:
+        year = d[0]
+        user_ucinetid = d[2]
+        grad_count = d[3]
+        grad_students = d[4]
+        previous_balance = d[5]
+
+        db = get_db()
+        db.execute("PRAGMA foreign_keys = ON")
+        user = get_exist_user(db, user_ucinetid)
+
+        if user is None:
+            print('User is not existed in the system') # TODO: display error to front-end
+            return
+
+        # TODO: Automatically calculate point after insert/update
+        user_id = user['user_id']
+        if action_type == 'addProfessors':
+            try:
+                insert_professors_point_info(db, user_id, year, previous_balance, grad_count, grad_students)
+            except db.IntegrityError:
+                print('INTEGRITY ERROR\n', traceback.print_exc())
+        elif action_type == 'editProfessors':
+            try:
+                update_professors_point_info(db, user_id, year, previous_balance, grad_count, grad_students)
+            except db.IntegrityError:
+                print('INTEGRITY ERROR\n', traceback.print_exc())
+    return
+
+def get_exist_user(db, user_ucinetid):
+    cur = db.cursor()
+    return cur.execute(
```

Comment:
Do we need cur variable? I think we can simply use db.execute() to execute sql statements

Reply:
I just realized we don't need cur since we don't have several operations that connect to the database at the same time. Updated to use single `db.execute()` command.